site/docs/how-to-release.md

```diff
@@ -324,7 +324,7 @@ Create a PR in the `iceberg` repo to make revapi run on the new release. For an
 
 ### Documentation Release
 
-Please follow the instructions on the GitHub repository in the [`README.md` in the `site/`](https://github.com/apache/iceberg/tree/main/site) directory.
```

Comment:
this deploy/release step should be executed after the doc and javadoc branches are updated/pushed first. moved it to the proper order later.